### PR TITLE
nodejs-week2

### PR DESCRIPTION
**Pull request by Sana-Shabeel**

### CODE REVIEW

**Comment by shpomp** [comment]
Good job overall! 🤩 
However, there are quite a few inconsistencies and some tasks not fully done. 
It is good that you are efficient and have more time to study after early HW submission, on the other hand, perhaps it would be beneficial to take a day or 2 longer but be much more thorough and consistent with the tasks. You are doing great and clearly have a very good grasp 💪🏽 , just a little rushed 🏃🏽 

This part of the homework is more or less overlooked:
> What if there is no meal with the requested id in meals.json?
> What if the users writes a string as id? fx "/api/meals/lol"
> What if the users write a query parameter that is not supported?

To summarize the specific comments, the returns and checks for user errors are quite inconsistent. In some cases, you don't check for those, in some you do, but then send different types of responses to client. I would ask you to review all of the endpoints and make sure the handling is there and it is consistent. 
And the `multiply` endpoint from warmup needs some love 🤓 

Once again, good job and good progress 👏🏽 , just some more attention to detail and consistency and it will be flawless 🏅  ⭐

**Review by shpomp — CHANGES_REQUESTED** [review]

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+router.get("/", async (request, response) => {
+  try {
+    response.send(reservations);
+  } catch (error) {
+    throw error;
+  }
+});
+router.get("/:id", async (req, res) => {
+  try {
+    if (req.params.id * 1 > reservations.length) {
```
[the remark this answers is not in the document]
Comment (Sana-Shabeel) [reply nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js:15]: converts to a number

File: nodejs/week2/homework/meal-sharing/controllers/mealsController.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  return res.send(
+    meals.filter(
+      (meal) => Date.parse(meal.createdAt) > Date.parse(createdAfter)
+    )
+  );
+};
+exports.getByLimit = async (req, res) => {
+  limit = Number(req.query.limit);
+  if (!limit) {
+    return res.status(400).send("Invalid limit");
```
[the remark this answers is not in the document]
Comment (Sana-Shabeel) [reply nodejs/week2/homework/meal-sharing/controllers/mealsController.js:48]: Thank you very much for the detailed review, i will. implement all the changes you suggested soon and also the not rushing part. the only reason i did that was to have time for side projects i was working on.

File: nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (shpomp) [reply nodejs/week2/homework/meal-sharing/src/backend/api/reservations-router.js:15]: I see, thanks!
Number() or parseInt() would be the way to go here.

File: nodejs/week2/homework/meal-sharing/controllers/mealsController.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (shpomp) [reply nodejs/week2/homework/meal-sharing/controllers/mealsController.js:48]: Understandable and great you are taking the initiative to do extra projects 💪🏽
